packages/user-operation-controller/src/UserOperationController.ts

```diff
@@ -147,24 +186,73 @@ export class UserOperationController extends BaseController<
    * @param request.to - Destination address of the resulting transaction.
    * @param request.value - Value to include in the resulting transaction.
    * @param options - Configuration options when creating a user operation.
-   * @param options.chainId - Chain ID of the resulting transaction.
+   * @param options.networkClientId - ID of the network client used to query the chain.
+   * @param options.origin - Origin of the user operation, such as the hostname of a dApp.
+   * @param options.requireApproval - Whether to require user approval before submitting the user operation. Defaults to true.
    * @param options.smartContractAccount - Smart contract abstraction to provide the contract specific values such as call data and nonce.
    */
   async addUserOperation(
-    request: {
-      data?: string;
-      maxFeePerGas: string;
-      maxPriorityFeePerGas: string;
-      to?: string;
-      value?: string;
-    },
-    options: { chainId: string; smartContractAccount: SmartContractAccount },
-  ) {
+    request: AddUserOperationRequest,
+    options: AddUserOperationOptions,
+  ): Promise<AddUserOperationResponse> {
     validateAddUserOperationRequest(request);
     validateAddUserOperationOptions(options);
 
-    const { chainId } = options;
-    const metadata = this.#createEmptyMetadata(chainId);
+    return await this.#addUserOperation(request, options);
+  }
+
+  /**
+   * Create and submit a user operation equivalent to the provided transaction.
+   *
+   * @param transaction - Transaction to use as the basis for the user operation.
+   * @param options - Configuration options when creating a user operation.
+   * @param options.networkClientId - ID of the network client used to query the chain.
+   * @param options.origin - Origin of the user operation, such as the hostname of a dApp.
+   * @param options.requireApproval - Whether to require user approval before submitting the user operation. Defaults to true.
+   * @param options.smartContractAccount - Smart contract abstraction to provide the contract specific values such as call data and nonce.
+   */
+  async addUserOperationFromTransaction(
+    transaction: TransactionParams,
+    options: AddUserOperationOptions,
+  ): Promise<AddUserOperationResponse> {
+    validateAddUserOperationOptions(options);
+
+    const { data, maxFeePerGas, maxPriorityFeePerGas, to, value } = transaction;
+
+    return await this.#addUserOperation(
+      {
+        data: data === '' ? undefined : data,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+        to,
+        value,
+      } as AddUserOperationRequest,
```

Comment:
Why is the purpose of this cast?

Reply:
This is no longer needed so removed now.

It was necessary initially as `maxFeePerGas` and `maxPriorityFeePerGas` weren't optional in the `AddUserOperationRequest` since we didn't support gas fee estimation, but were optional in the `TransactionParams`.

Both optional now.